private/buf/bufprint/curated_plugin_printer.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufprint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strconv"
+
+	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+)
+
+type curatedPluginPrinter struct {
+	writer io.Writer
+}
+
+func newCuratedPluginPrinter(writer io.Writer) *curatedPluginPrinter {
+	return &curatedPluginPrinter{
+		writer: writer,
+	}
+}
+
+func (p *curatedPluginPrinter) PrintCuratedPlugin(_ context.Context, format Format, plugin *registryv1alpha1.CuratedPlugin) error {
+	switch format {
+	case FormatText:
+		return p.printCuratedPluginsText(plugin)
+	case FormatJSON:
+		return json.NewEncoder(p.writer).Encode(
+			registryCuratedPluginToOutputCuratedPlugin(plugin),
+		)
+	default:
+		return fmt.Errorf("unknown format: %v", format)
+	}
+}
+
+func (p *curatedPluginPrinter) PrintCuratedPlugins(_ context.Context, format Format, nextPageToken string, plugins ...*registryv1alpha1.CuratedPlugin) error {
+	switch format {
+	case FormatText:
+		return p.printCuratedPluginsText(plugins...)
+	case FormatJSON:
+		outputPlugins := make([]outputCuratedPlugin, 0, len(plugins))
+		for _, plugin := range plugins {
+			outputPlugins = append(outputPlugins, registryCuratedPluginToOutputCuratedPlugin(plugin))
+		}
+		return json.NewEncoder(p.writer).Encode(paginationWrapper{
+			NextPage: nextPageToken,
+			Results:  outputPlugins,
+		})
+	default:
+		return fmt.Errorf("unknown format: %v", format)
+	}
+}
+
+func (p *curatedPluginPrinter) printCuratedPluginsText(plugins ...*registryv1alpha1.CuratedPlugin) error {
+	if len(plugins) == 0 {
+		return nil
+	}
+	return WithTabWriter(
+		p.writer,
+		[]string{
+			"Owner",
+			"Name",
+			"Version",
+			"Revision",
+		},
+		func(tabWriter TabWriter) error {
+			for _, plugin := range plugins {
+				if err := tabWriter.Write(
+					plugin.Owner,
+					plugin.Name,
+					plugin.Version,
+					strconv.FormatInt(int64(plugin.Revision), 10),
+				); err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+	)
+}
+
+type outputCuratedPlugin struct {
+	Owner    string `json:"owner"`
+	Name     string `json:"name"`
+	Version  string `json:"version"`
+	Revision uint32 `json:"revision"`
```

Comment:
Other useful fields might be the description, image digest, etc. For now this will print the basics.

@amckinney - Is it ok if the JSON representation prints more details than the tab formatted one? In particular the image digest is going to be pretty long - Docker supports this by truncating image digest by default and providing a `--no-trunc` flag to show the full digest.

Reply:
I'll let Alex be the authority on this one, but from my experience the tabbed output is usually a subset of the json output. 

So imo it's fine to expose the minimum in the tabbed output, and if a user needs additional data they reach for the json representation (typically piped into `jq` or whatever).

Reply:
I agree with @mfridman - it's fine for the JSON representation to be a superset (it often is).

For what it's worth, the default output could just be the fully-qualified name of the plugin (without any tab-delimited format), similar to the `buf push` command. That way it's immediately visible how the user could reference the plugin they just created in a `buf.gen.yaml`, `buf.plugin.yaml`, etc. For example, something like this (`buf push` is shown for clarity):

```
$ buf push
7e040f51304842f6b55fdea26225ac16
```

```
$ buf alpha plugin push
buf.build/library/go:v1.0.0-1
```

This is fine for now though - given that we still don't exactly know how the revision will be included at the end of the fully-qualified plugin reference, it's probably safer to start with this.